examples/addsvc/client/grpc/factory.go

```diff
@@ -7,33 +7,42 @@ import (
 
 	"github.com/go-kit/kit/endpoint"
 	"github.com/go-kit/kit/examples/addsvc/pb"
+	"github.com/go-kit/kit/loadbalancer"
+	kitot "github.com/go-kit/kit/tracing/opentracing"
 	grpctransport "github.com/go-kit/kit/transport/grpc"
+	"github.com/opentracing/opentracing-go"
 )
 
 // SumEndpointFactory transforms GRPC host:port strings into Endpoints that call the Sum method on a GRPC server
 // at that address.
-func SumEndpointFactory(instance string) (endpoint.Endpoint, io.Closer, error) {
-	cc, err := grpc.Dial(instance, grpc.WithInsecure())
-	return grpctransport.NewClient(
-		cc,
-		"Add",
-		"Sum",
-		encodeSumRequest,
-		decodeSumResponse,
-		pb.SumReply{},
-	).Endpoint(), cc, err
+func NewSumEndpointFactory(tracer opentracing.Tracer) loadbalancer.Factory {
+	return func(instance string) (endpoint.Endpoint, io.Closer, error) {
+		cc, err := grpc.Dial(instance, grpc.WithInsecure())
+		return grpctransport.NewClient(
+			cc,
+			"Add",
+			"Sum",
+			encodeSumRequest,
+			decodeSumResponse,
+			pb.SumReply{},
+			grpctransport.SetClientBefore(kitot.ToGRPCRequest(tracer)),
```

Comment:
the only "real" changes in this file are these before funcs... the rest is just indenting.